energy-dao/src/external_sc_interactions/farm_actions.rs

```diff
@@ -0,0 +1,161 @@
+multiversx_sc::imports!();
+
+use common_structs::PaymentsVec;
+use farm::{
+    base_functions::ClaimRewardsResultType, EnterFarmResultType, ExitFarmWithPartialPosResultType,
+};
+use locked_token_wrapper::wrapped_token;
+
+use super::farm_config::{FarmState, WrappedFarmTokenAttributes, MAX_PERCENT};
+
+#[multiversx_sc::module]
+pub trait FarmActionsModule:
+    crate::external_sc_interactions::farm_config::FarmConfigModule
+    + crate::external_sc_interactions::locked_token_actions::LockedTokenModule
+    + utils::UtilsModule
+    + energy_query::EnergyQueryModule
+    + lkmex_transfer::energy_transfer::EnergyTransferModule
+    + legacy_token_decode_module::LegacyTokenDecodeModule
+    + wrapped_token::WrappedTokenModule
+    + simple_lock::token_attributes::TokenAttributesModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    fn call_enter_farm(
+        &self,
+        farm_address: ManagedAddress,
+        farming_tokens: PaymentsVec<Self::Api>,
+    ) -> EnterFarmResultType<Self::Api> {
+        self.farm_proxy(farm_address)
+            .enter_farm_endpoint(OptionalValue::<ManagedAddress>::None)
+            .with_multi_token_transfer(farming_tokens)
+            .execute_on_dest_context()
+    }
+
+    fn call_exit_farm(
+        &self,
+        farm_address: ManagedAddress,
+        farm_tokens: EsdtTokenPayment,
+    ) -> ExitFarmWithPartialPosResultType<Self::Api> {
+        self.farm_proxy(farm_address)
+            .exit_farm_endpoint(
+                farm_tokens.amount.clone(),
+                OptionalValue::<ManagedAddress>::None,
+            )
+            .with_esdt_transfer(farm_tokens)
+            .execute_on_dest_context()
+    }
+
+    fn call_farm_claim(
+        &self,
+        farm_addr: ManagedAddress,
+        farm_token: EsdtTokenPayment,
+    ) -> ClaimRewardsResultType<Self::Api> {
+        self.farm_proxy(farm_addr)
+            .claim_rewards_endpoint(OptionalValue::<ManagedAddress>::None)
+            .with_esdt_transfer(farm_token)
+            .execute_on_dest_context()
+    }
+
+    fn update_farm_after_claim(
+        &self,
+        initial_farm_state: &FarmState<Self::Api>,
+        farm_state_mapper: &mut SingleValueMapper<FarmState<Self::Api>>,
+        new_farm_token: EsdtTokenPayment,
+        farm_rewards: EsdtTokenPayment,
+        division_safety_constant: &BigUint,
+    ) {
+        farm_state_mapper.update(|config| {
```

Comment:
This code is a bit big for a lambda. I'd say rewrite it as:
```rust
let config = farm_state_mapper.get();

/* your code */

farm_state_mapper.set(config);
```

Since `update` does the same thing anyway.

Reply:
Updated.